### PR TITLE
New Rule Proj2000: Embed valid resource files

### PR DESCRIPTION
**Pull request by Corniel**
# Proj2000: Embed valid resources
A resource file should be a valid RESX file.

## Compliant
``` XML
<?xml version="1.0" encoding="utf-8"?>
<root>
  <resheader name="resmimetype">
    <value>text/microsoft-resx</value>
  </resheader>
  <resheader name="reader">
    <value>System.Resources.ResXResourceReader, System.Windows.Forms</value>
  </resheader>
  <resheader name="writer">
    <value>System.Resources.ResXResourceWriter, System.Windows.Forms</value>
  </resheader>
  <data name="King" xml:space="preserve">
    <value>Koning</value>
  </data>
  <data name="Queen" xml:space="preserve">
    <value>Koningin</value>
  </data>
</root>
```

### CODE REVIEW

File: rules/Proj2000.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Proj2000: Embed valid resources
+A resource file should be a valid RESX file.
+
+## Compliant
+``` XML
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (LauraXiulan) [inline rules/Proj2000.md:6]: How much data should be in the RESX file to be regarded as compliant? Would just a resheader be enough or does it have to be specific values?

File: src/DotNetProjectFile.Analyzers/Rule.cs [hunk 2]
```diff
@@ -137,6 +137,16 @@ public static class Rule
         category: Category.CodeQuality,
         isEnabled: true);
 
+    public static DiagnosticDescriptor EmbedValidResourceFiles => New(
+        id: 2000,
+        title: "Embed valid resource files",
+        message: "Invalid XML: {0}",
+        description: "A resource file should contain valid XML.",
```
Comment (LauraXiulan) [inline src/DotNetProjectFile.Analyzers/Rule.cs:144]: Its not totally clear if it is about content or structure

File: rules/Proj2000.md (hunk 1 above)
Comment (Corniel) [reply rules/Proj2000.md:6]: I've to check explicitly which headers are required. I considered to add that (and the root node name) as a requirement, but for now this should be sufficient.

File: src/DotNetProjectFile.Analyzers/Rule.cs (hunk 2 above)
Comment (Corniel) [reply src/DotNetProjectFile.Analyzers/Rule.cs:144]: Because it is about both?!

**Review by LauraXiulan — APPROVED** [review]

**Pull request merged**